lib/task_logger.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+require "json"
+require "time"
+
+class TaskLogger
+  # Initialize logger with default parameters
+  def initialize( task, identifier, type, subtype )
+    @task = task
+    @identifier = identifier || default_identifier( task )
+    @type = type
+    @subtype = subtype
+    @start_time = nil
+    @log_file = File.join( CONFIG.task_logger_output, "#{task}.log" )
```

Comment:
We'll probably want to handle the case where `CONFIG.task_logger_output` is undefined. I don't think we'd want it to throw an error - we likely want people to be able to run the code w/o needing to define the config. Alternatively we could do what we're doing with the [application logstash logs](https://github.com/inaturalist/inaturalist/blob/main/config/initializers/logstasher.rb#L25) and put them in the rails log directory which we know will exist. This to me seems preferred that way all the logs are in the same place, we don't need an extra config, there will be different logs for different environments, and we know we'll be able to create a task_logger log file and write to it.

Reply:
if I'm not using the "default" folder to task logs, I need to update the filebeat config, which is currently only monitoring this default folder (`/var/log/inaturalist-tasks/*.log`) 

I can for sure add a second path like:
```
paths:
    - /var/log/inaturalist-tasks/*.log
    - /home/inaturalist/deployment/production/current/log/task_log_*.logstash.log
```

I suppose it won't impact anything else, since Ruby filebeat config is looking at:
`/home/inaturalist/deployment/production/current/log/production.logstash.log`

is it what you had in mind?

Reply:
That is what I had in mind. To me, it would be preferred that the logs generated by this application all go to the same place - `$RAILS_ROOT/logs/*.log`. I understand that other task logs will go elsewhere, but since those will be "generated" externally to this codebase, e.g. in bash or elsewhere, it makes sense they would live in a different location